doc/netplan.md

```diff
@@ -1007,6 +1008,60 @@ Examples:
 
 :   Alternate name for the ``key`` field. See above.
 
+Wireguard-specific keys:
+
+    ``private-key`` (scalar) – since **0.100**
+    :   Base64-encoded private key, prefixed with ``base64:``. Or an absolute
+        path to a file with the private key.
+
+    ``fwmark`` (scalar) – since **0.100**
+    :   Firewall mark for outgoing WireGuard packets from this interface,
+        optional.
+
+    ``port`` (scalar) – since **0.100**
+    :   UDP port to listen at or ``auto``. Optional, defaults to ``auto``.
+
+    ``peers`` (sequence of mappings) – since **0.100**
+    :   A list of peers, each having keys documented below.
+
+    Example:
+
+    tunnels:
+        wireguard0:
+            mode: wireguard
+            peers:
+                - public-key: /some/public.key
+                  allowed-ips: [0.0.0.0/0, "2001:fe:ad:de:ad:be:ef:1/24"]
+                  keepalive: 23
+                  remote: 1.2.3.4:5
+                - public-key: /other/public.key
+                  allowed-ips: [0.0.0.0/0, "10.10.10.20/24"]
+                  keepalive: 22
+                  remote: 5.4.3.2:1
+
+    ``public-key`` (scalar) – since **0.100**
+    :   Peer's base64-encoded public key, required.
```

Comment:
There are 2 levels of keys for wireguard:
* **private key**: defined directly for the tunnel interface (e.g. `wg0`) – basically the "input-key" decrypting incoming packages
* **public & shared keys**: defined for each individual peer (remote endpoint) of that tunnel interface

I've rebuilt the wireguard key handling so that on the tunnel level one can specify the `key:` (private key) as scalar (or use `keys.input-key` respectively), while on the wireguard peers level one can use the `key:` scalar to define the public key (always required) or optionally define the `keys.public:` & `keys.shared` mapping if an optional shared key shall be defined:

Example1:
```
    tunnels:
      wg0:
        mode: wireguard
        addresses: [...]
        key: mNb7OIIXTdgW4khM7OFlzJ+UPs7lmcWHV7xjPgakMkQ=
        peers:
          - keys:
              public: rlbInAj0qV69CysWPQY7KEBnKxpYCpaWqOs/dLevdWc=
              shared: 7voRZ/ojfXgfPOlswo3Lpma1RJq7qijIEEUEMShQFV8=
            ...
```

Example2:
```
    tunnels:
        wg0:
            mode: wireguard
            keys:
                input-key: /some/private.key
            peers:
                - key: rlbInAj0qV69CysWPQY7KEBnKxpYCpaWqOs/dLevdWc=
                  allowed-ips: [0.0.0.0/0, "2001:fe:ad:de:ad:be:ef:1/24"]
                  keepalive: 23
                  remote: 1.2.3.4:5
                - keys:
                      public: M9nt4YujIOmNrRmpIRTmYSfMdrpvE7u6WkG8FY8WjG4=
                      shared: /some/shared.key
                  allowed-ips: [10.10.10.20/24]
                  keepalive: 22
                  remote: 5.4.3.2:1
```